Wsign-conversion appears on latest master when building with Clang

Currently on latest master (`6b74da4757a549563d7c37c8fae3e704662a043b`) the following warning appears when building with Clang (tested on 14.0.0):

```bash
root@9cdd6f96b60d:/# clang++ -I googletest/googletest/include -I googletest/googletest -Wconversion -c googletest/googletest/src/gtest-all.cc 
In file included from googletest/googletest/src/gtest-all.cc:41:
googletest/googletest/src/gtest.cc:4082:33: warning: implicit conversion changes signedness: 'const char' to 'unsigned char' [-Wsign-conversion]
        if (IsValidXmlCharacter(ch)) {
            ~~~~~~~~~~~~~~~~~~~ ^~
googletest/googletest/src/gtest.cc:4083:56: warning: implicit conversion changes signedness: 'const char' to 'unsigned char' [-Wsign-conversion]
          if (is_attribute && IsNormalizableWhitespace(ch))
                              ~~~~~~~~~~~~~~~~~~~~~~~~ ^~
googletest/googletest/src/gtest.cc:4104:29: warning: implicit conversion changes signedness: 'const char' to 'unsigned char' [-Wsign-conversion]
    if (IsValidXmlCharacter(*it))
        ~~~~~~~~~~~~~~~~~~~ ^~~
```

The fix should be simple and it seems it's already done in some places, like line 4084:

```cpp
            m << "&#x" << String::FormatByte(static_cast<unsigned char>(ch))
```

--- a/googletest/src/gtest.cc
+++ b/googletest/src/gtest.cc
@@ -4079,8 +4079,9 @@ std::string XmlUnitTestResultPrinter::EscapeXml(
           m << '"';
         break;
       default:
-        if (IsValidXmlCharacter(ch)) {
-          if (is_attribute && IsNormalizableWhitespace(ch))
+        if (IsValidXmlCharacter(static_cast<unsigned char>(ch))) {
+          if (is_attribute && IsNormalizableWhitespace(
+                                            static_cast<unsigned char>(ch)))
             m << "&#x" << String::FormatByte(static_cast<unsigned char>(ch))
               << ";";
           else
@@ -4101,7 +4102,7 @@ std::string XmlUnitTestResultPrinter::RemoveInvalidXmlCharacters(
   std::string output;
   output.reserve(str.size());
   for (std::string::const_iterator it = str.begin(); it != str.end(); ++it)
-    if (IsValidXmlCharacter(*it))
+    if (IsValidXmlCharacter(static_cast<unsigned char>(*it)))
       output.push_back(*it);
 
   return output;

